File: reactor/r2dbc-mysql/src/main/java/ua/mai/zyme/r2dbcmysql/config/AppUtil.java
package ua.mai.zyme.r2dbcmysql.config;

import reactor.core.publisher.Mono;
import ua.mai.zyme.r2dbcmysql.entity.Balance;

import java.time.LocalDateTime;
import java.util.concurrent.ExecutionException;

public class AppUtil {

    static public LocalDateTime now() {
        return roundLocalDateTime(LocalDateTime.now());
    }
    static public LocalDateTime roundLocalDateTime(LocalDateTime dateTime) {
        return dateTime.withNano(0);
    }

    static public Balance getBalanceFromMono(Mono<Balance> mono) {
        try {
            return mono.toFuture().get();
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        } catch (ExecutionException e) {
            throw new RuntimeException(e);
        }
    }



}

File: kotlin/kotlin-r2dbc-mysql/kotlin-r2dbc-mysql-be/src/main/java/ua/mai/zyme/r2dbcmysql/util/AppUtil.kt
package ua.mai.zyme.r2dbcmysql.util

import reactor.core.publisher.Mono
import ua.mai.zyme.r2dbcmysql.entity.Balance
import java.time.LocalDateTime
import java.util.concurrent.ExecutionException

object AppUtil {

    @JvmStatic
    fun now(): LocalDateTime {
        return roundLocalDateTime(LocalDateTime.now())
    }

    @JvmStatic
    fun roundLocalDateTime(dateTime: LocalDateTime): LocalDateTime {
        return dateTime.withNano(0)
    }

    @JvmStatic
    fun getBalanceFromMono(mono: Mono<Balance>): Balance {
        return try {
            mono.toFuture().get()
        } catch (e: InterruptedException) {
            throw RuntimeException(e)
        } catch (e: ExecutionException) {
            throw RuntimeException(e)
        }
    }
}